client/object_put.go

```diff
@@ -356,17 +356,45 @@ func (x *payloadWriter) Close() error {
 // future. Be ready to refactor your code regarding imports and call mechanics,
 // in essence the operation will not change.
 func CreateObject(ctx context.Context, cli *Client, signer neofscrypto.Signer, cnr cid.ID, owner user.ID, data io.Reader, attributes ...string) (oid.ID, error) {
-	s, err := NewDataSlicer(ctx, cli, signer, cnr, owner)
+	if len(attributes)%2 != 0 {
+		return oid.ID{}, slicer.ErrInvalidAttributeAmount
+	}
+
+	s, err := NewDataSlicer(ctx, cli)
 	if err != nil {
 		return oid.ID{}, err
 	}
 
+	netInfo, err := cli.NetworkInfo(ctx, PrmNetworkInfo{})
+	if err != nil {
+		return oid.ID{}, fmt.Errorf("read current network info: %w", err)
+	}
+
+	var opts slicer.Options
+	opts.SetObjectPayloadLimit(netInfo.MaxObjectSize())
+	opts.SetCurrentNeoFSEpoch(netInfo.CurrentEpoch())
+	if !netInfo.HomomorphicHashingDisabled() {
+		opts.CalculateHomomorphicChecksum()
+	}
+
 	var hdr object.Object
 	hdr.SetContainerID(cnr)
+	hdr.SetCreationEpoch(netInfo.CurrentEpoch())
```

Comment:
`netInfo.CurrentEpoch()` is called the second time in that func

BTW, do we need to have epoch in the params if we set it here? or, vice versa: do we need to set it here if we have `opts.SetCurrentNeoFSEpoch`?

Reply:
You are right, `hdr.SetCreationEpoch(netInfo.CurrentEpoch())` is not required here, removed